### DISCUSSION TITLE
Hourly forecast should include relative humidity

### DISCUSSION THREAD

**Discussion by brewin** [weather-gov/api#368] [Q&A] [answered]
The /forecast/hourly endpoint doesn't currently include relative humidity. It's impossible to calculate a heat index without it. The old hourly forecast API does include it.

**Comment by scadergit** [comment]
Sorry for the delayed reply. This appears to be an oversight. We'll determine when it can be added back in, but that might take longer than expected due to proceses outside of our control. As noted elsewhere, the forecast endpoints are not heavily prioroitized at this time.

**Comment by dmeinrath** [comment]
any update on this?

**Comment by scadergit** [comment]
Sorry, only more of this:
> that might take longer than expected due to proceses outside of our control.

**Comment by jcpowell** [comment]
Any update on this?

**Reply by scadergit** [reply, marked as answer]
The API patch that adds dewpoint, chance of precip, and relative humidity has passed local testing. From what I hear, that will be released in the next few months.

**Comment by zim2dive** [comment]
Very much looking forward to precip % being added to the hourly forecast! thanks!

**Comment by hjcheck** [comment]
Any update on this, @scadergit ? I am still not seeing relative humidity on the hourly forecast API, and as @brewin said, you cannot calculate a heat index without it. Thank you.

**Reply by sullynole** [reply]
> Any update on this, @scadergit ? I am still not seeing relative humidity on the hourly forecast API, and as @brewin said, you cannot calculate a heat index without it. Thank you.

Do you have an example url?  Relative humidity should be available on the hourly forecast endpoint when populated.

**Reply by hjcheck** [reply]
In a way, it was available, but I ended up having to parse out _relativeHumidity_ because it isn't directly available as a parameter (how I was preferably looking for the data). The below is how I was able to get what I needed through R.

For anyone else who may be struggling with it and you are coming across this thread like I did, I've attached my R script, so you can just copy/paste if it helps.

~ Replace "{wfo}" with the 3 digit Weather Forecasting Office code relative to your area.
~ Replace "x,y" with your respective latitude and longitude.

_library(httr)
res <- VERB("GET", url = "https:// api.weather.gov/gridpoints/**{wfo}**/**x,y**/forecast/hourly?units=us")
cat(content(res, 'text'))
new <- fromJSON(content(res, as = "text", encoding = "UTF-8"), flatten = TRUE)
NOAAtable <- tibble(
  pulldate=new$properties$generatedAt,
  forecastdate=new$properties$periods$startTime,
  temperature=new$properties$periods$temperature,
  windSpeed=new$properties$periods$windSpeed,
  humidity=new$properties$periods$relativeHumidity.value) %>%
  mutate(pulldate=ymd_hms(pulldate, tz = "America/Chicago"),
         forecastdate=ymd_hms(forecastdate, tz = "America/Chicago"))_
